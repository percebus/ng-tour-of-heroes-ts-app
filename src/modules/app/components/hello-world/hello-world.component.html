<div class="hello-world">
  <h1>Hello {{ name }}!</h1>
</div>
