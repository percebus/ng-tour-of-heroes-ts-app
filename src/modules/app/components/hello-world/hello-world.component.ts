import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { Tentative } from '../../types/nothing';

@Component({
  selector: 'app-hello-world',
  templateUrl: './hello-world.component.html',
  styleUrls: ['./hello-world.component.scss'],
})
export class HelloWorldComponent implements OnInit {
  name: Tentative<string>;

  constructor(private route: ActivatedRoute) {}

  refresh() {
    const name = this.route.snapshot.paramMap.get('name') ?? 'World';
    this.name = name;
  }

  ngOnInit(): void {
    this.refresh();
  }
}
