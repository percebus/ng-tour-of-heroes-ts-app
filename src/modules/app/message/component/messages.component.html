<div class="messages-component" *ngIf="messageService.messages.length">
  <h2>Messages</h2>
  <button type="button" class="clear" (click)="clear()">Clear messages</button>
  <div *ngFor="let message of messageService.messages">
    {{ message }}
  </div>
</div>
