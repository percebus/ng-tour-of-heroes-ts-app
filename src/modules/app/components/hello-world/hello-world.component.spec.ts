import { FormsModule } from '@angular/forms';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HelloWorldComponent } from './hello-world.component';
import { ActivatedRoute, Router } from '@angular/router';
import { RouterTestingModule } from '@angular/router/testing';

describe('HelloWorldComponent', () => {
  let oHelloWorldComponent: HelloWorldComponent;
  let oComponentFixture: ComponentFixture<HelloWorldComponent>;
  let oRouter: Router;
  let oActivatedRoute: ActivatedRoute;

  describe('rendering', () => {
    describe('by default', () => {
      beforeEach(() => {
        TestBed.configureTestingModule({
          imports: [FormsModule, RouterTestingModule.withRoutes([])],
          declarations: [HelloWorldComponent],
        });

        oRouter = TestBed.inject(Router);
        oActivatedRoute = TestBed.inject(ActivatedRoute);

        oComponentFixture = TestBed.createComponent(HelloWorldComponent);
        oComponentFixture.detectChanges();
        oHelloWorldComponent = oComponentFixture.componentInstance;
      });

      it('is instanceof HelloWorld', () => {
        expect(oHelloWorldComponent).toBeInstanceOf(HelloWorldComponent);
      });

      it('displays "Hello World!"', () => {
        const oHTMLElement = oComponentFixture.nativeElement //
          .querySelector('.hello-world h1');

        expect(oHTMLElement.textContent).toEqual('Hello World!');
      });

      it('matches snapshot', () => {
        expect(oComponentFixture).toMatchSnapshot();
      });
    });

    describe('name: "David"', () => {
      beforeEach(() => {
        const mockActivatedRoute = {
          snapshot: {
            paramMap: {
              get: () => 'David',
            },
          },
        };

        TestBed.configureTestingModule({
          imports: [FormsModule, RouterTestingModule.withRoutes([])],
          declarations: [HelloWorldComponent],
          providers: [
            { provide: ActivatedRoute, useValue: mockActivatedRoute },
          ],
        });

        oRouter = TestBed.inject(Router);
        oActivatedRoute = TestBed.inject(ActivatedRoute);

        oComponentFixture = TestBed.createComponent(HelloWorldComponent);
        oComponentFixture.detectChanges();
        oHelloWorldComponent = oComponentFixture.componentInstance;
      });

      it('displays "Hello David!"', () => {
        const oHTMLElement = oComponentFixture.nativeElement //
          .querySelector('.hello-world h1');

        expect(oHTMLElement.textContent).toEqual('Hello David!');
      });

      it('matches snapshot', () => {
        expect(oComponentFixture).toMatchSnapshot();
      });
    });
  });
});
