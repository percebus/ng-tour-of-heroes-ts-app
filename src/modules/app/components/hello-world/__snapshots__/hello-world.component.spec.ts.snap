// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorldComponent rendering by default matches snapshot 1`] = `
<app-hello-world
  name={[Function String]}
  route={[Function ActivatedRoute]}
>
  <div
    class="hello-world"
  >
    <h1>
      Hello World!
    </h1>
  </div>
</app-hello-world>
`;

exports[`HelloWorldComponent rendering name: "David" matches snapshot 1`] = `
<app-hello-world
  name={[Function String]}
  route={[Function Object]}
>
  <div
    class="hello-world"
  >
    <h1>
      Hello David!
    </h1>
  </div>
</app-hello-world>
`;
